src/main/fc/rc_modes.c

```diff
@@ -125,3 +125,27 @@ bool isModeActivationConditionPresent(boxId_e modeId)
 
     return false;
 }
+
+void removeModeActivationCondition(const boxId_e modeId)
+{
+    unsigned offset = 0;
+    for (unsigned i = 0; i < MAX_MODE_ACTIVATION_CONDITION_COUNT; i++) {
+        modeActivationCondition_t *mac = modeActivationConditionsMutable(i);
+
+        if (mac->modeId == modeId && !offset) {
+            offset++;
+        }
+
+        if (offset) {
```

Comment:
I would reduce indentation level of following large block by making it `continue` prematurely :)

Reply:
I like it the way it is - early wraparound does not improve readability in my opinion, unless the following block is overly long (which in most cases should really be addressed by splitting it up into functions.